executor/main.go

```diff
@@ -62,38 +94,57 @@ func main() {
 			if file == nil {
 				msg := "Received nil input file on the executor"
 				errors.WriteResponse(rs, http.StatusBadRequest, msg, nil)
+				logger.Error().
+					Msg(msg)
 				return nil
 			}
 			testFileName := fmt.Sprintf("inputs/%s", file.Filename)
 			testFile, testFileErr := os.Create(testFileName)
 			if testFileErr != nil {
 				msg := "An error occurred while trying to create a local file named '" + file.Filename + "' on 'inputs/' folder"
 				errors.WriteResponse(rs, http.StatusBadRequest, msg, testFileErr)
+				logger.Error().
+					Err(testFileErr).
+					Msg(msg)
 				return nil
 			}
 			defer testFile.Close()
 			receivedTestFile, rfErr := file.Open()
 			if rfErr != nil {
 				msg := "An error occurred while trying to open the received test file named '" + file.Filename + "'"
 				errors.WriteResponse(rs, http.StatusBadRequest, msg, rfErr)
+				logger.Error().
+					Err(rfErr).
+					Msg(msg)
 				return nil
 			}
 			defer receivedTestFile.Close()
 			_, copyErr := io.Copy(testFile, receivedTestFile)
 			if copyErr != nil {
 				msg := "An error occurred while trying to copy the received test to a local file named '" + file.Filename + "' on 'inputs/' folder"
 				errors.WriteResponse(rs, http.StatusInternalServerError, msg, copyErr)
+				logger.Error().
+					Err(copyErr).
+					Msg(msg)
 				return nil
 			}
 		}
 
-		res := executor.Execute("program.out", "inputs", 2.)
+		res := executor.Execute("program.out", "inputs", 2., logger)
 		jsonResult, convertErr := json.Marshal(res)
 		if convertErr != nil {
 			msg := "An error occurred while trying to convert the execution result into a json format"
 			errors.WriteResponse(rs, http.StatusInternalServerError, msg, convertErr)
+			logger.Error().
+					Err(convertErr).
+					Msg(msg)
 			return nil
 		}
+		jsonResult, err := json.Marshal(res)
```

Comment:
The JSON Marshal already happened, this should be inside the above if, checking convertErr.

Reply:
We were actually running the marshall twice, probably because it got duplicated during a merge. We removed the second marshall run so this should be resolved now.